kitsune/users/models.py

```diff
@@ -458,7 +464,72 @@ class AccountEvent(models.Model):
                                                   default=None,
                                                   null=True,
                                                   blank=False)
-    fxa_uid = models.CharField(blank=True, null=True, unique=True, max_length=128)
+    fxa_uid = models.CharField(blank=True, null=True, max_length=128)
     jwt_id = models.CharField(max_length=256)
     issued_at = models.CharField(max_length=32)
     profile = models.ForeignKey(Profile, related_name='account_events', null=True)
+    error = models.TextField(max_length=4096, blank=True)
+
+    def process(self):
+        if self.status == self.PROCESSED:
+            return
+
+        try:
+            if self.event_type == self.DELETE_USER:
+                self.profile.delete()
```

Comment:
Profiles are not deleted but rather deactivated which purges any PII from the db. Profile deletion would also affect the AAQ, forums, KB etc